client/my-sites/checkout/src/components/checkout-sidebar-plan-upsell/index.tsx

```diff
@@ -16,10 +16,76 @@ import {
 	getItemVariantCompareToPrice,
 	getItemVariantDiscountPercentage,
 } from '../item-variation-picker/util';
+import type { WPCOMProductVariant } from '../item-variation-picker';
 import './style.scss';
 
 const debug = debugFactory( 'calypso:checkout-sidebar-plan-upsell' );
 
+function getUpsellVariant( currentVariant: WPCOMProductVariant, variants: WPCOMProductVariant[] ) {
+	// Return the binennial plan if the plan in cart is a monthly or annual plan
+	if (
+		currentVariant.productBillingTermInMonths === 1 ||
+		currentVariant.productBillingTermInMonths === 12
+	) {
+		return variants?.find( ( product ) => product.termIntervalInMonths === 24 );
+	}
+
+	// Return the triennial plan if the plan in cart is a binennial plan
+	if ( currentVariant.productBillingTermInMonths === 24 ) {
+		return variants?.find( ( product ) => product.termIntervalInMonths === 36 );
+	}
+
+	debug( 'plan in cart has no upsell variant; variants are', variants );
+}
+
+function getUpsellTextForVariant(
+	upsellVariant: WPCOMProductVariant,
+	percentSavings: number,
+	__: any
+) {
+	if ( upsellVariant.productBillingTermInMonths === 24 ) {
```

Comment:
Similar to above, we could probably dedupe this code by associating the billing term's number of months with the expected text output, then interpolate them into the translation function. But, more of a nit than anything.

Reply:
To clarify, are you suggesting that we change the translation to something like this?
```js
sprintf( __( '%(year)s-year cost' ), { year: 2 } )
```

Reply:
I was thinking about how we could perhaps have some sort of associative array of termsInMonths to text to display, but I think you're right about the readability of the file, so the way you have it currently looks better on second look. Marking resolved 👍